src/dev/beta.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import * as nls from 'vscode-nls'
+import * as vscode from 'vscode'
+import * as AdmZip from 'adm-zip'
+import got from 'got'
+import globals from '../shared/extensionGlobals'
+import { getLogger } from '../shared/logger'
+import { VSCODE_EXTENSION_ID } from '../shared/extensions'
+import { makeTemporaryToolkitFolder } from '../shared/filesystemUtilities'
+import { reloadWindowPrompt } from '../shared/utilities/vsCodeUtils'
+import { isUserCancelledError, ToolkitError } from '../shared/errors'
+import { SystemUtilities } from '../shared/systemUtilities'
+import { telemetry } from '../shared/telemetry/telemetry'
+import { cast } from '../shared/utilities/typeConstructors'
+import { CancellationError } from '../shared/utilities/timeoutUtils'
+
+const localize = nls.loadMessageBundle()
+
+const downloadIntervalMs = 1000 * 60 * 60 * 24 // A day in milliseconds
+const betaToolkitKey = 'dev.beta'
+
+interface BetaToolkit {
+    readonly needUpdate: boolean
+    readonly lastCheck: number
+}
+
+function getBetaToolkitData(vsixUrl: string): BetaToolkit | undefined {
+    return globals.context.globalState.get<Record<string, BetaToolkit>>(betaToolkitKey, {})[vsixUrl]
+}
+
+async function updateBetaToolkitData(vsixUrl: string, data: BetaToolkit) {
+    await globals.context.globalState.update(betaToolkitKey, {
+        ...globals.context.globalState.get<Record<string, BetaToolkit>>(betaToolkitKey, {}),
+        [vsixUrl]: data,
+    })
+}
+
+/**
+ * Watch the beta VSIX daily for changes.
+ * If this is the first time we are watching the beta version or if its been 24 hours since it was last checked then try to prompt for update
+ */
+export function watchBetaVSIX(vsixUrl: string): vscode.Disposable {
```

Comment:
should we add a `once()` log an INFO or WARN message here so that the logs have positive confirmation that this codepath was entered?

Reply:
Added a log statement. `once` isn't needed as this function is only executed once.